lib/Service/DeleteService.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Nextcloud - ContextChat
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Anupam Kumar <kyteinsky@gmail.com>
+ * @copyright Anupam Kumar 2024
+ */
+
+declare(strict_types=1);
+
+namespace OCA\ContextChat\Service;
+
+use OCA\ContextChat\BackgroundJobs\DeleteJob;
+use OCA\ContextChat\Db\QueueDelete;
+use OCA\ContextChat\Db\QueueDeleteMapper;
+use OCA\ContextChat\Type\DeleteContext;
+use OCP\BackgroundJob\IJobList;
+
+class DeleteService {
+	public function __construct(
+		private IJobList $jobList,
+		private QueueDeleteMapper $deleteMapper,
+	) {
+	}
+
+	/**
+	 * @param string $type
+	 * @param string $payload
+	 * @param string $userId
+	 * @return void
+	 */
+	public function scheduleDelete(string $type, string $payload, string $userId = ''): void {
+		$item = new QueueDelete();
+		$item->setType($type);
+		$item->setUserId($userId);
+		$item->setPayload($payload);
+
+		// do not catch DB exceptions
+		$this->deleteMapper->insertIntoQueue($item);
+
+		if (!$this->jobList->has(DeleteJob::class, null)) {
+			$this->jobList->add(DeleteJob::class, null);
+		}
+	}
+
+	/**
+	 * @param array<QueueDelete> $entities
+	 * @return array
```

Comment:
Would be good to type the return type properly for psalm magic :)

Reply:
yeah I tried, psalm said the type in the comment was too specific.